src/components/map/mapbox/layers/ClusterLayer.tsx

```diff
@@ -13,20 +21,44 @@ import {
   fetchResultNoStore,
 } from "../../../common/store/searchReducer";
 import { centroid } from "@turf/turf";
-import { GeoJSONSource, MapLayerMouseEvent } from "mapbox-gl";
+import { GeoJSONSource, MapLayerMouseEvent, Popup } from "mapbox-gl";
 import { useDispatch } from "react-redux";
 import { AppDispatch } from "../../../common/store/store";
+import { createRoot } from "react-dom/client";
+import MapPopup from "../popup/MapPopup";
+import { CircularProgress } from "@mui/material";
 
 interface ClusterLayerProps {
   // Vector tile layer should added to map
   collections: Array<OGCCollection>;
   // Event fired when user click on the point layer
   onDatasetSelected?: (uuid: Array<string>) => void;
+  onClickPopup?: (uuid: string) => void;
 }
 
 const OPACITY = 0.6;
 const STROKE_WIDTH = 1;
 
+// Constants for cluster circle sizes
+//These define the radius(px) of the circles used to represent clusters on the map.
+const DEFAULT_CIRCLE_SIZE = 20;
+const MEDIUM_CIRCLE_SIZE = 30;
+const LARGE_CIRCLE_SIZE = 40;
+const EXTRA_LARGE_CIRCLE_SIZE = 60;
+
+// Constants for cluster circle colors
+// These define the colors used for the circles representing clusters of different sizes.
+const DEFAULT_COLOR = "#51bbd6";
+const MEDIUM_COLOR = "#f1f075";
+const LARGE_COLOR = "#f28cb1";
+const EXTRA_LARGE_COLOR = "#fe8cf1";
+
+// Constants for point count thresholds
+// These define the boundaries between different cluster sizes.
+const MEDIUM_POINT_COUNT = 20;
+const LARGE_POINT_COUNT = 30;
+const EXTRA_LARGE_POINT_COUNT = 50;
+
```

Comment:
these contants is very specific to this layer so it makes sense to put it there but grouping it is good to make it more readable, it will be good to have those to be default props of the component so user can change it via props